src/coap_blockwise.c

```diff
@@ -142,9 +142,9 @@ static enum golioth_status process_blockwise_uploads(struct golioth_client *clie
 {
     enum golioth_status status = GOLIOTH_ERR_FAIL;
     int ret = call_read_block_callback(ctx);
```

Comment:
This function does actually return a negative value, so the assert you hit must be caused by something else.

Reply:
Indeed, I tested this as working but if I return `GOLIOTH_OK` it asserts:

```
    if (block_idx == 10)
    {
        *block_size = 0;
        *is_last = true;
        return GOLIOTH_OK;
    }
```